### PR TITLE
fixes #24121 , documentation update distributed caching

### LINKED ISSUE #24121: .NET 6: Distributed caching in ASP.NET Core
V6 code was added in #24363
Now just update the doc to use the new code

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0d01025d-86b2-bb2d-86c1-ef0695bc98e3
* Version Independent ID: 5b9a6572-3671-0d54-b2ef-83d04f8b3e96
* Content: [Distributed caching in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/performance/caching/distributed?view=aspnetcore-6.0)
* Content Source: [aspnetcore/performance/caching/distributed.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/performance/caching/distributed.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-performance**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by smandia**
Fixes #24121

***EDIT by @Rick-Anderson**
[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/performance/caching/distributed?view=aspnetcore-6.0&branch=pr-en-us-24400)

### CODE REVIEW

**Comment by smandia** [comment]
@Rick-Anderson - Document updated . 
Curious/Stupid question - Can you please double check line - **There's need to use a Singleton or Scoped lifetime for** I updated it to say there is need . Is it correct statement or we have other good ways to do things.
@fiyazbinhasan

File: aspnetcore/performance/caching/distributed.md [hunk 1]
```diff
@@ -226,7 +226,7 @@ The <xref:Microsoft.Extensions.Caching.Distributed.IDistributedCache> interface
 
 ## Establish distributed caching services
 
-Register an implementation of <xref:Microsoft.Extensions.Caching.Distributed.IDistributedCache> in `Startup.ConfigureServices`. Framework-provided implementations described in this topic include:
+Register an implementation of <xref:Microsoft.Extensions.Caching.Distributed.IDistributedCache> in `Program.cs`. Framework-provided implementations described in this topic include:
```
Comment (Rick-Anderson) [inline aspnetcore/performance/caching/distributed.md:229]: Revert all the changes  from here down.  You only make changes to the the .NET 6 version between
```
::: moniker range=">= aspnetcore-6.0"
// make changes here
::: moniker-end
```

Everything below `::: moniker range="< aspnetcore-6.0"` is for the versions lower than 6.

File: aspnetcore/performance/caching/distributed.md [hunk 2]
```diff
@@ -142,22 +142,22 @@ To configure NCache:
...
 Each time the Index page is loaded, the cache is checked for the cached time in `OnGetAsync`. If the cached time hasn't expired, the time is displayed. If 20 seconds have elapsed since the last time the cached time was accessed (the last time this page was loaded), the page displays *Cached Time Expired*.
 
 Immediately update the cached time to the current time by selecting the **Reset Cached Time** button. The button triggers the `OnPostResetCachedTime` handler method.
 
-[!code-csharp[](distributed/samples/3.x/DistCacheSample/Pages/Index.cshtml.cs?name=snippet_IndexModel&highlight=7,14-20,25-29)]
+[!code-csharp[](distributed/samples/6.x/DistCacheSample/Pages/Index.cshtml.cs?name=snippet_IndexModel&highlight=7,14-20,25-29)]
 
 > [!NOTE]
-> There's no need to use a Singleton or Scoped lifetime for <xref:Microsoft.Extensions.Caching.Distributed.IDistributedCache> instances (at least for the built-in implementations).
+> There's need to use a Singleton or Scoped lifetime for <xref:Microsoft.Extensions.Caching.Distributed.IDistributedCache> instances.
```
Comment (Rick-Anderson) [inline aspnetcore/performance/caching/distributed.md:160]:
```suggestion
There's  ***no*** need to use a Singleton or Scoped lifetime for <xref:Microsoft.Extensions.Caching.Distributed.IDistributedCache> instances with the built-in implementations.
```

File: aspnetcore/performance/caching/distributed.md [hunk 3]
```diff
@@ -142,22 +142,22 @@ To configure NCache:
...
 The sample app injects <xref:Microsoft.Extensions.Caching.Distributed.IDistributedCache> into the `IndexModel` for use by the Index page.
 
 Each time the Index page is loaded, the cache is checked for the cached time in `OnGetAsync`. If the cached time hasn't expired, the time is displayed. If 20 seconds have elapsed since the last time the cached time was accessed (the last time this page was loaded), the page displays *Cached Time Expired*.
 
 Immediately update the cached time to the current time by selecting the **Reset Cached Time** button. The button triggers the `OnPostResetCachedTime` handler method.
 
-[!code-csharp[](distributed/samples/3.x/DistCacheSample/Pages/Index.cshtml.cs?name=snippet_IndexModel&highlight=7,14-20,25-29)]
+[!code-csharp[](distributed/samples/6.x/DistCacheSample/Pages/Index.cshtml.cs?name=snippet_IndexModel&highlight=7,14-20,25-29)]
 
 > [!NOTE]
```
Comment (Rick-Anderson) [inline aspnetcore/performance/caching/distributed.md:159]:
```suggestion
```

**Review by Rick-Anderson — CHANGES_REQUESTED** [review]

**Review by Rick-Anderson — APPROVED** [review]

File: aspnetcore/performance/caching/distributed.md [hunk 4]
```diff
@@ -129,10 +129,10 @@ To configure NCache:
 
 1. Install [NCache open source NuGet](https://www.nuget.org/packages/Alachisoft.NCache.OpenSource.SDK/).
 1. Configure the cache cluster in [client.ncconf](https://www.alachisoft.com/resources/docs/ncache/admin-guide/client-config.html).
-1. Add the following code to `Startup.ConfigureServices`:
+1. Add the following code to `Program.cs`:
 
    ```csharp
-   services.AddNCacheDistributedCache(configuration =>    
+   builder.Services.AddNCacheDistributedCache(configuration =>
```
Comment (Rick-Anderson) [inline aspnetcore/performance/caching/distributed.md:135]: Move this snippet to `Program.cs`. Be sure to test the program.

**Comment by smandia** [comment]
@Rick-Anderson  - Please have a look again

File: aspnetcore/performance/caching/distributed.md (hunk 4 above)
Comment (smandia) [reply aspnetcore/performance/caching/distributed.md]: done .

File: aspnetcore/performance/caching/distributed.md (hunk 3 above)
Comment (smandia) [reply aspnetcore/performance/caching/distributed.md]: done

File: aspnetcore/performance/caching/distributed.md (hunk 2 above)
Comment (smandia) [reply aspnetcore/performance/caching/distributed.md]: done

File: aspnetcore/performance/caching/distributed.md (hunk 1 above)
Comment (smandia) [reply aspnetcore/performance/caching/distributed.md]: done

**Review by Rick-Anderson — APPROVED** [review]

**Comment by Rick-Anderson** [comment]
@smandia  nice work, thanks.

**Pull request merged**